### ISSUE TITLE
Trained for 16 epochs with batch size 8, but it seems like it's not converging

### ISSUE THREAD

**Issue by binbinsh** [ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild#2]
Dear author, thank you very much for the codes, I have tried to train the model on LRS2, I keep all parameters except for the batch size remained unchanged. The model was run on RTX4090 and the batch size was changed to 8 to fit into the memory. However, the results seem incorrect. could you please help me take a look? The training log is as follows after 16 epochs:

```
******** Training [16 / 200] : 64808 / 142157, Iter Time : 0.366 sec, Learning Rate of 0.000100 ********
######## Step(Epoch): 292405(16), Recon Loss: 0.8434956669807434 #########
VP: S
GT: HEAL'S MADE IT POSSIBLE FOR THAT

VP: THE
GT: BARONET'S BEDROOM AGAIN JUST AS IT WAS WHEN THE NATIONAL TRUST TOOK OVER THE PROPERTY

VP: S
GT: WHAT DID CHARLES HANSON SPEND THE 1 ⁇ 5 ON CHARLES I REALLY DUG

Cutting Video frames off. Requires 276 frames: pretrain/5673096029702830518/00023. But Max_timestep is 250
Cutting Video frames off. Requires 274 frames: pretrain/6225908571815403760/00020. But Max_timestep is 250
******** Training [16 / 200] : 65608 / 142157, Iter Time : 0.349 sec, Learning Rate of 0.000100 ********
######## Step(Epoch): 292505(16), Recon Loss: 1.0348305702209473 #########
VP: THE
GT: AT TODAY'S PRICES THE NET ADDITIONAL BENEFIT WOULD BE AROUND  ⁇

VP: S
GT: IT SEEMED TO WORK BETTER THAN TEACHING I WASN'T A NATURAL TEACHER WHEN

VP: S
GT: TO LEARN FROM THEIR PARENTS AND THAT'S WHAT THE LONG CHILDHOOD IS ABOUT IF
```

And the visualization in tensorboard looks like this:

<img width="1177" alt="tb" src="https://github.com/ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild/assets/95557195/f00d9d02-26ef-4a3c-96d2-9f538efc7c00">

Can you help me take a look and identify what might be causing the issue? Or is this a normal occurrence?

**Comment by ms-dot-k** [comment]
Hello, 
It seems that the model cannot go into the optimization point. It is usually seen when we train lip reading models with CTC loss.
We need to initialize the visual front-end with the pre-trained model with lip reading loss.

I have modified the `train.py` to be able to load the pre-trained visual front-end by using the argument `--visual_front_checkpoint`.
Also, I have uploaded the pre-trained visual front-end. Please find it on the updated readme.md.

**Comment by binbinsh** [comment]
Hello ms-dot-k,

Thank you very much for the visual front-end model, I tried using the pre-trained visual front-end you provided as per your suggestion. I successfully ran the training script with the `--visual_front_checkpoint` argument, and the results for the first step seemed promising. However, when I reached step 101, I encountered the same issues with the output, as shown below:

```
******** Training [0 / 200] : 8 / 142157, Iter Time : 0.008 sec, Learning Rate of 0.000100 ********
######## Step(Epoch): 1(0), Recon Loss: 1.1831997632980347 #########
VP: AMERICA STROKE HADN STROKE AMERICA STROKE AMERICA THROUGH HADN FLEW BAT FLEW IDEAL TECHNOLOGICAL HADN NURSE AMERICA DIVORCE LAB DID RECIPE SCREEN IDEAL NURSE HADN AMERICAMI HADN DID ROOM MOUNTAIN NURSE GENOME STROKE DAD DID NURSERR INDIGENOUS SIGN AMERICA REACH FLEW HADN FLEW AMERICA THROUGH NURSE STROKE HUNG IDEAL AMERICA INDIGENOUS AMERICA NURSE JUSTICE HADN AMERICA NURSE HUNG HADNJA AMERICA MAL DID THROUGH INDIGENOUS AMERICA NURSE LAB REACH STROKE NURSE TECHNOLOGICAL QUALITIES NURSE SCREEN AMERICA HADN SCREEN NURSE SCREEN AMERICA SACRED REACH SACREDACTIVE NURSE RELATE BAT STORE AMERICA STROKE AMERICA FLEW
GT: ONE THAT RUNS ABOUT IN SHORTS TELLING YOU HOW THE

VP: AUSTRALIA MID INDIGENOUS RACING C SAND C SAND C LAB SAND VICTOR DUE BREED AWESOME MID C DAILY VICTOR C SAND VICTOR C VICTOR SAND DAILY SAND INDIGENOUS EARTHQUAKE C DAILY SPECTACULAR VICTOR RACING SAND VICTOR C VICTOR C VICTOR CHAM SAND AWESOME C VICTOR SAND VICTOR AWESOME SAND C SAND DAILY SAND VICTOR DAILY VICTOR C REVOLUTION DUE C VICTOR C AWESOME VICTOR SAND DAILY EARTHQUAKE DAILY VICTOR SAND VICTOR SAND RELIGION C
GT: GOT A SORT OF RUGGED HONEST APPEAL

VP: NURSE HONOR NURSECOM NURSE 500 NURSE CRAFT NURSED NURSE 500ACE NURSE EQUIVALENT NURSE CRAFT NURSE HONOR NURSE POTENTIALLY HONOR ELEPHANT NURSE HONOR NURSEACE ELEPHANT NURSE DELICIOUS NURSE 500 NURSE 500 NURSE ELEPHANT NURSE CRAFT HONOR NURSE CLASSIC NURSE ELEPHANT NURSE HONOR NURSE STRESS NURSE REASONS NURSE CRAFTD NURSE 500 NURSE 500 CLO NURSE ELEPHANT DELICIOUS ELEPHANT ANTIQUES NURSEARIES CRAFT RENEWABLE
GT: I CAN'T PROVE THERE'S NOT AN INVISIBLE UNICORN WELDING A TURD TO THE ROOF OF THIS BUILDING

Cutting Video frames off. Requires 264 frames: pretrain/5958143560340715925/00047. But Max_timestep is 250
******** Training [0 / 200] : 808 / 142157, Iter Time : 1.708 sec, Learning Rate of 0.000100 ********
######## Step(Epoch): 101(0), Recon Loss: 0.7637051939964294 #########
VP: S
GT: AND I KNEW IF I DID ALL RIGHT

VP:
GT: JUST DOES NOT FLOAT YOUR BOAT IN ANY WAY IT WOULD AT THE

VP:
GT: THESE STUDIES WOULD PROVE VITAL WHEN IT CAME TO DEVELOPING NEW VISION
```

As you can see, the model's output (VP) became the same as without pretrained visual front-end model. I am not sure what might be causing this issue, and I would greatly appreciate your insights and guidance on how to resolve it.

**Comment by ms-dot-k** [comment]
It only initializes the ResNet part so it does not directly produce the nice words.
I confirmed that it starts to find optimization points after 6k steps using a batch size of 24.

Maybe, you can use a learning rate scheduler with a warmup for better training.
